### ISSUE TITLE
Lazy reflection of a set

### ISSUE THREAD

**Issue by mforets** [JuliaReach/LazySets.jl#1558]
Given a convex set `X`, its opposite with respect to the origin is the set `-X := {-x : x  ∈ X}`. This issue proposes to create a lazy oposite, `Opposite(X)` ( `Reflection` / point reflection wrt the origin).

**Comment by ueliwechsler** [comment]
Would this conceptually correspond to just defining a new type
```julia
struct Reflection
    X
end
```
and the overloading of the support value and vector as:
`support_value(d, R::Reflection) = support_value(-d, R.X)`
`support_vector(d, R::Reflection) = -support_vector(-d, R.X)`

**Comment by schillic** [comment]
> Would this conceptually correspond to just defining a new type
> 
> ```julia
> struct Reflection
>     X
> end
> ```
> 
> and the overloading of the support value and vector as:
> `support_value(d, R::Reflection) = support_value(-d, R.X)`
> `support_vector(d, R::Reflection) = -support_vector(-d, R.X)`

Yes, and other operations are also straightforward, such as membership or `linear_map`.
